### PR TITLE
Created Settings page tests

### PR DESCRIPTION
**Pull request by marinaprozorovska**

### CODE REVIEW

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs [hunk 1]
```diff
@@ -18,9 +18,34 @@ public void SetUp()
...
+
+            IWebElement dateFormatField = driver.FindElement(By.Id("dateFormat"));
+            IWebElement numberFormatField = driver.FindElement(By.Id("numberFormat"));
+            IWebElement defaultCurrencyField = driver.FindElement(By.Id("currency"));
+            IWebElement saveButton = driver.FindElement(By.Id("save"));
+            SelectElement dateFormatDropdown = new SelectElement(dateFormatField);
+            SelectElement numberFormatDropdown = new SelectElement(numberFormatField);
+            SelectElement defaultCurrencyDropdown = new SelectElement(defaultCurrencyField);
+
+            new WebDriverWait(driver, TimeSpan.FromSeconds(200))
```
Comment (varajan) [inline CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs]: 200 seconds is too much.
```suggestion
            new WebDriverWait(driver, TimeSpan.FromSeconds(10))
```

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs [hunk 2]
```diff
@@ -18,9 +18,34 @@ public void SetUp()
...
+             .Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementToBeClickable(By.Id("save")));
+            dateFormatDropdown.SelectByText("dd/MM/yyyy");
+            numberFormatDropdown.SelectByText("123,456,789.00");
+            defaultCurrencyDropdown.SelectByText("$ - US dollar");
+            saveButton.Click();
+            System.Threading.Thread.Sleep(2000);
```
Comment (varajan) [inline CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs]: Try to replace the 'Thread.Sleep' with Wait. It should be such condition, as 'wait for alert is shown'

File: CalculatorTests/CalculatorTests/Tests/SettingsPageTests.cs [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        private IWebDriver driver;
+
+        [SetUp]
+        public void SetUp()
+        {
+            var options = new ChromeOptions { AcceptInsecureCertificates = true };
+            driver = new ChromeDriver(options);
+            driver.Url = "https://localhost:5001/Calculator";
+
+
```
Comment (varajan) [inline CalculatorTests/CalculatorTests/Tests/SettingsPageTests.cs]: Please remove the empty lines.

File: CalculatorTests/CalculatorTests/Tests/SettingsPageTests.cs [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+       [TestCase("dd-MM-yyyy", "123.456.789,00", "€ - euro", "€", "100.000,00", "200.000,00", "19-05-2023")]
+       [TestCase("MM/dd/yyyy", "123 456 789.00", "£ - Great Britain Pound", "£", "100 000.00", "200 000.00", "05/19/2023")]
+       [TestCase("MM dd yyyy", "123 456 789,00", "₴ - Ukrainian hryvnia", "₴", "100 000,00", "200 000,00", "05 19 2023")]
+
+
+        public void SettingsTests (string dateFormat, string numberFormat, string defaultCurrency, string amountCurrency, string interestEarned, string income, string endDate)
+        {
+            IWebElement settingsButton = driver.FindElement(By.XPath("//div[@class = 'settings link btn btn-link']"));
+          
+
```
Comment (varajan) [inline CalculatorTests/CalculatorTests/Tests/SettingsPageTests.cs]: Please leave only one empty line.

File: CalculatorTests/CalculatorTests/Tests/SettingsPageTests.cs [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+       [TestCase("dd/MM/yyyy", "123,456,789.00", "$ - US dollar", "$", "100,000.00", "200,000.00", "19/05/2023")]
+       [TestCase("dd-MM-yyyy", "123.456.789,00", "€ - euro", "€", "100.000,00", "200.000,00", "19-05-2023")]
+       [TestCase("MM/dd/yyyy", "123 456 789.00", "£ - Great Britain Pound", "£", "100 000.00", "200 000.00", "05/19/2023")]
+       [TestCase("MM dd yyyy", "123 456 789,00", "₴ - Ukrainian hryvnia", "₴", "100 000,00", "200 000,00", "05 19 2023")]
+
+
+        public void SettingsTests (string dateFormat, string numberFormat, string defaultCurrency, string amountCurrency, string interestEarned, string income, string endDate)
+        {
+            IWebElement settingsButton = driver.FindElement(By.XPath("//div[@class = 'settings link btn btn-link']"));
```
Comment (varajan) [inline CalculatorTests/CalculatorTests/Tests/SettingsPageTests.cs]: You can avoid using the full class name:
```suggestion
            IWebElement settingsButton = driver.FindElement(By.XPath("//div[contains(@class, 'settings') ]"));
```

File: CalculatorTests/CalculatorTests/Tests/SettingsPageTests.cs [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+            SelectElement dateFormatDropdown = new SelectElement(dateFormatField);
+            SelectElement numberFormatDropdown = new SelectElement(numberFormatField);
+            SelectElement defaultCurrencyDropdown = new SelectElement(defaultCurrencyField);
+
+
+            dateFormatDropdown.SelectByText(dateFormat);
+            numberFormatDropdown.SelectByText(numberFormat);
+            defaultCurrencyDropdown.SelectByText(defaultCurrency);
+            saveButton.Click();
+            System.Threading.Thread.Sleep(2000);
```
Comment (varajan) [inline CalculatorTests/CalculatorTests/Tests/SettingsPageTests.cs]: Please replace 'Thread.Sleep' with explicit wait.

File: CalculatorTests/CalculatorTests/Tests/SettingsPageTests.cs [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+            SelectElement dateFormatDropdown = new SelectElement(dateFormatField);
+            SelectElement numberFormatDropdown = new SelectElement(numberFormatField);
+            SelectElement defaultCurrencyDropdown = new SelectElement(defaultCurrencyField);
+
+            dateFormatDropdown.SelectByText(dateFormat);
+            numberFormatDropdown.SelectByText(numberFormat);
+            defaultCurrencyDropdown.SelectByText(defaultCurrency);
+            cancelButton.Click();
+            
+            IWebElement depositAmountField = driver.FindElement(By.XPath("//tr[contains( string(), 'Deposit Amount')]//input"));
```
Comment (varajan) [inline CalculatorTests/CalculatorTests/Tests/SettingsPageTests.cs]: I suggest to go back to the Settings page and verify the values in dropdowns are equal to default ones.

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs (hunk 1 above)
Comment (marinaprozorovska) [reply CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs]: Fixed

File: CalculatorTests/CalculatorTests/Tests/SettingsPageTests.cs (hunk 3 above)
Comment (marinaprozorovska) [reply CalculatorTests/CalculatorTests/Tests/SettingsPageTests.cs]: Done

File: CalculatorTests/CalculatorTests/Tests/SettingsPageTests.cs (hunk 4 above)
Comment (marinaprozorovska) [reply CalculatorTests/CalculatorTests/Tests/SettingsPageTests.cs]: Done

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs (hunk 2 above)
Comment (marinaprozorovska) [reply CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs]: Fixed

File: CalculatorTests/CalculatorTests/Tests/SettingsPageTests.cs (hunk 5 above)
Comment (marinaprozorovska) [reply CalculatorTests/CalculatorTests/Tests/SettingsPageTests.cs]: Fixed

File: CalculatorTests/CalculatorTests/Tests/SettingsPageTests.cs (hunk 6 above)
Comment (marinaprozorovska) [reply CalculatorTests/CalculatorTests/Tests/SettingsPageTests.cs]: Fixed

File: CalculatorTests/CalculatorTests/Tests/SettingsPageTests.cs (hunk 7 above)
Comment (marinaprozorovska) [reply CalculatorTests/CalculatorTests/Tests/SettingsPageTests.cs]: Done

**Review by varajan — APPROVED** [review]

**Pull request merged**